bindings/wallet-core/src/wallet.rs

```diff
@@ -135,24 +144,61 @@ impl Wallet {
         self.icarus.dump_in(&mut dump);
 
         let (ignored, transactions) = dump.finalize();
+        let mut raws = Vec::with_capacity(transactions.len());
 
-        let transactions = transactions
-            .into_iter()
-            .map(|t| t.serialize_as_vec().unwrap())
-            .collect();
+        for (used, f) in transactions {
+            let id = f.id();
+
+            self.pending_transactions.insert(id, used);
+            raws.push(f.serialize_as_vec().unwrap());
+        }
 
         Conversion {
             ignored,
-            transactions,
+            transactions: raws,
+        }
+    }
+
+    /// use this function to confirm a transaction has been properly received
+    ///
+    /// This function will automatically update the state of the wallet
+    pub fn confirm_transaction(&mut self, id: FragmentId) {
```

Comment:
shouldn't this error if the transactions is not pending/was already confirmed?

Reply:
good point